### PR TITLE
Map templates are now globally unit tested.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Should identify specific details of #20. Will also tack on a commit to fix any identified map failures to this PR.

### CODE REVIEW

**Comment by afterthought2** [comment]
Suggest placing them during init, before the testing starts, so that map tests will run on them consistently. Not sure why it hangs.

**Comment by PsiOmegaDelta** [comment]
It's probably preferable to load any map templates in `/datum/controller/subsystem/unit_tests/proc/load_map_templates()` for this reason.

File: code/_helpers/logging.dm [hunk 1]
```diff
@@ -199,11 +199,23 @@
...
-	return "[..()] ([t ? t : "NULL"]) ([t ? "[t.x],[t.y],[t.z]" : "0,0,0"]) ([t ? t.type : "NULL"])"
+	if(t)
+		var/obj/effect/overmap/visitable/O = map_sectors["[z]"]
+		if(istype(O))
+			return "[..()] ([t]) ([t.x],[t.y],[t.z] - [O.name]) ([t.type])"
+		else
+			return "[..()] ([t]) ([t.x],[t.y],[t.z]) ([t.type])"
+
+	else
+		return "[..()] (NULL) ([t ? "[t.x],[t.y],[t.z]" : "0,0,0"]) (NULL)"
```
Comment (PsiOmegaDelta) [inline code/_helpers/logging.dm]: `	return "[..()] (NULL) ("0,0,0") (NULL)"`
The `else` can be dropped as well, the if-statement above will never fall through as is.

File: code/controllers/subsystems/mapping.dm [hunk 2]
```diff
@@ -32,6 +32,17 @@ SUBSYSTEM_DEF(mapping)
...
+/datum/controller/subsystem/mapping/proc/includeTemplate(var/datum/map_template/map_template, var/list/banned_maps)
+	if(!initial(map_template.id))
+		return FALSE
+	var/datum/map_template/MT = new map_template()
+	if(banned_maps)
+		for(var/mappath in MT.mappaths)
+			if(banned_maps.Find(mappath))
+				return FALSE
+	map_templates[MT.name] = MT
+	. = MT
```
Comment (PsiOmegaDelta) [inline code/controllers/subsystems/mapping.dm]: This proc is mixing returning reference and value types (a template or `FALSE`)
Should either return instance/`null` or `TRUE`/`FALSE`

Comment (PsiOmegaDelta) [reply code/controllers/subsystems/mapping.dm]: Line 43-44 should probable be replaced with `return TRUE`

https://github.com/NebulaSS13/Nebula/pull/22/files#diff-c2e5790416598e06d4b9215742ea8c70R58-R59 should be replaced with
```
if(includeTemplate(item, banned_maps)
	map_templates[MT.name] = MT
	. = MT
else
	continue
```
That way https://github.com/NebulaSS13/Nebula/pull/22/files#diff-de1774595a6a972d2ecda2c7215a5aacR83-R86 can simply be changed to
```
/datum/controller/subsystem/mapping/includeTemplate(var/datum/map_template/map_template, var/banned_maps)
	return TRUE
```

Comment (PsiOmegaDelta) [inline code/controllers/subsystems/mapping.dm:44]: This proc is mixing returning reference and value types (map template and `FALSE`)
Should either return the map template and null or `TRUE`/`FALSE`

Comment (MistakeNot4892) [reply code/controllers/subsystems/mapping.dm:44]: Removed `FALSE` return values on earlier returns.

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**